pyocd/rtos/threadx.py

```diff
@@ -0,0 +1,514 @@
+    # subtracted.
+    NOFPU_REGISTER_OFFSETS = {
+        # Software stacked
+        -1:     0,      # lr (exception)
+        4:      4,      # r4
+        5:      8,      # r5
+        6:      12,     # r6
+        7:      16,     # r7
+        8:      20,     # r8
+        9:      24,     # r9
+        10:     28,     # r10
+        11:     32,     # r11
+        # Hardware stacked
+        0:      36,     # r0
+        1:      40,     # r1
+        2:      44,     # r2
+        3:      48,     # r3
+        12:     52,     # r12
+        14:     56,     # lr (thread)
+        15:     60,     # pc
+        16:     64,     # xpsr
+    }
+
+    # Cortex-m0 port of threadx reverses r4-7 and r8-11
+    NOFPU_REGISTER_OFFSETS_V6M = {
+        # Software stacked
+        4:      20,     # r4
+        5:      24,     # r5
+        6:      28,     # r6
+        7:      32,     # r7
+        8:      4,      # r8
+        9:      8,      # r9
+        10:     12,     # r10
+        11:     16,     # r11
+    }
+
+    FPU_REGISTER_OFFSETS = {
+        # Software stacked
+        -1:     0,      # lr (exception)
+        0x50:   4,      # s16
+        0x51:   8,      # s17
+        0x52:   12,     # s18
+        0x53:   16,     # s19
+        0x54:   20,     # s20
+        0x55:   24,     # s21
+        0x56:   28,     # s22
+        0x57:   32,     # s23
+        0x58:   36,     # s24
+        0x59:   40,     # s25
+        0x5a:   44,     # s26
+        0x5b:   48,     # s27
+        0x5c:   52,     # s28
+        0x5d:   56,     # s29
+        0x5e:   60,     # s30
+        0x5f:   64,     # s31
+        4:      68,     # r4
+        5:      72,     # r5
+        6:      76,     # r6
+        7:      80,     # r7
+        8:      84,     # r8
+        9:      88,     # r9
+        10:     92,     # r10
+        11:     96,     # r11
+        # Hardware stacked
+        0:      100,    # r0
+        1:      104,    # r1
+        2:      108,    # r2
+        3:      112,    # r3
+        12:     116,    # r12
+        14:     120,    # lr
+        15:     124,    # pc
+        16:     128,    # xpsr
+        0x40:   132,    # s0
+        0x41:   136,    # s1
+        0x42:   140,    # s2
+        0x43:   144,    # s3
+        0x44:   148,    # s4
+        0x45:   152,    # s5
+        0x46:   156,    # s6
+        0x47:   160,    # s7
+        0x48:   164,    # s8
+        0x49:   168,    # s9
+        0x4a:   172,    # s10
+        0x4b:   176,    # s11
+        0x4c:   180,    # s12
+        0x4d:   184,    # s13
+        0x4e:   188,    # s14
+        0x4f:   192,    # s15
+        33:     196,    # fpscr
+        # (reserved word: 200)
+    }
+
+    def __init__(self, parent, thread):
+        super(ThreadXThreadContext, self).__init__(parent)
+        self._thread = thread
+        self._has_fpu = self.core.has_fpu
+        self.nofpu_register_offsets = self.NOFPU_REGISTER_OFFSETS
```

Comment:
Good catch, thanks - the case you make is not really far fetched, as e.g. NXP LPC43S37 has two M0 (+?) and one M4 and if a soft abi is used for FP, it will appear just as you envisioned.